contracts/test/MaliciousAccount.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+import "../interfaces/IAccount.sol";
+import "../interfaces/IEntryPoint.sol";
+
+contract MaliciousAccount is IAccount {
+    IEntryPoint private ep;
+    constructor(IEntryPoint _ep) payable {
+        ep = _ep;
+    }
+    function validateUserOp(UserOperation calldata userOp, bytes32, address, uint256 missingAccountFunds)
+    external returns (uint256 sigTimeRange) {
+        ep.depositTo{value : missingAccountFunds}(address(this));
+        uint256 requiredGas = userOp.callGasLimit + userOp.verificationGasLimit + userOp.preVerificationGas;
+        uint256 gasPrice = missingAccountFunds / requiredGas;
+        uint256 basefee = gasPrice - userOp.maxPriorityFeePerGas;
+        require (basefee == userOp.maxPriorityFeePerGas, "Revert after first validation");
```

Comment:
I don't understand these last 2  lines:
you're checking that gasPrice = 2*maxPriorityFeePerGas
looking at this contract, it is completely unclear why, and why it is "malicious" in doing so.

Reply:
Where do I check that gasPrice = 2*maxPriorityFeePerGas?

I explicitly check that  basefee == userOp.maxPriorityFeePerGas, which is just a controlled parameter that any wallet can set to signal something on-chain.